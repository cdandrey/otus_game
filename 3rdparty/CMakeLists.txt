cmake_minimum_required(VERSION 3.15)

# Header-only libraries do not require compilation; simply mark them as INTERFACE
add_subdirectory(expected)

# Collect all header file directories from nested folders.
set(OTG_3RDPARTY_INCLUDE_DIRECTORIES
    "${CMAKE_CURRENT_SOURCE_DIR}/expected"
    # If you have more, add them here.
    # "${CMAKE_CURRENT_SOURCE_DIR}/что-то-ещё"
    PARENT_SCOPE
)