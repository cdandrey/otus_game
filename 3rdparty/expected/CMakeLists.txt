cmake_minimum_required(VERSION 3.2)

add_library(Expected
    STATIC 
    expected.h
)

target_include_directories(Expected
	PUBLIC
	"${CMAKE_CURRENT_SOURCE_DIR}"
)

target_link_libraries(Expected OtgLib)